Content.Server/CharacterInfo/RenameOnSpawnSystem.cs

```diff
@@ -0,0 +1,107 @@
+using Content.Server.Access.Systems;
+using Content.Server.Administration;
+using Content.Server.Administration.Systems;
+using Content.Server.Mind.Components;
+using Content.Server.PDA;
+using Content.Server.Station.Systems;
+using Content.Server.StationRecords;
+using Content.Shared.Access.Components;
+using Content.Shared.PDA;
+using Content.Shared.StationRecords;
+using Robust.Server.GameObjects;
+using Robust.Server.Player;
+using Robust.Shared.Enums;
+
+namespace Content.Server.CharacterInfo
+{
+    public sealed class RenameOnSpawnSystem : EntitySystem
+    {
+        [Dependency] private readonly IdCardSystem _idCardSystem = default!;
+        [Dependency] private readonly QuickDialogSystem _quickDialog = default!;
+        [Dependency] private readonly AdminSystem _adminSystem = default!;
+        [Dependency] private readonly PDASystem _pdaSystem = default!;
+        [Dependency] private readonly StationRecordsSystem _stationRecords = default!;
+        [Dependency] private readonly StationSystem _stationSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<RenameOnSpawnComponent, PlayerAttachedEvent>(OnPlayerAttached);
+        }
+
+        private void OnPlayerAttached(EntityUid uid, RenameOnSpawnComponent component, PlayerAttachedEvent args)
+        {
+            ShowNameChangePopup(component, args.Player);
+        }
+
+        public void ShowNameChangePopup(RenameOnSpawnComponent component, IPlayerSession player)
+        {
+            //popup rename menu
+            _quickDialog.OpenDialog(player, "Rename yourself", "Name", (string newName) =>
```

Comment:
You're going to need to make quickdialog look less ass if you want to use it here.